src/components/users/userProfile/EditUserProfile.js

```diff
@@ -0,0 +1,110 @@
+import TextField from "@mui/material/TextField";
+import Button from "@mui/material/Button";
+import React from "react";
+import { useState, useEffect } from "react";
+import client from "../../../utils/client";
+import Header from "../../Header/Header";
+import { useParams } from "react-router-dom";
+import "./style.css";
+
+const UserForm = ({ role }) => {
+  const initialFormData = {
+    firstName: "",
+    lastName: "",
+    email: "",
+    password: "",
+    biography: "",
+    githubUrl: "",
+  };
+
+  const [profile, setProfile] = useState(initialFormData);
+  const [editUserResponse, setEditUserResponse] = useState("");
+
+  const { id } = useParams();
+
+  useEffect(() => {
+    client
+      .get(`/user/${id}`)
+      .then((res) => {
+        setProfile(res.data.data.user);
+      })
+      .catch((err) => console.log(err.response));
+  }, [id]);
+
+  const handleChange = (event) => {
+    event.preventDefault();
+    const { value, name } = event.target;
+
+    setProfile({
+      ...profile,
+      [name]: value,
+    });
+  };
+
+  const onSubmit = (e) => {
+    e.preventDefault();
+    const putBody = {
+      firstName: profile.firstName,
+      lastName: profile.lastName,
+      email: profile.email,
```

Comment:
Email and password are not part of the form I think, so probably better to remove them from the state object and the out request.

Reply:
Good point Mike, think I included that by default but you're right, it's not needed in this state. Removing it now